### PR TITLE
Added payload session

### PR DESCRIPTION
**Pull request by tynyttie**
## Summary:
I updated sso_logging_info function to include the session identifier, this is used by many controllers to track user behavior.

## Related issue(s)
department-of-veterans-affairs/va.gov-team#0000
Team: identity-kong
VET_API:
Ticket: [issues/gh/department-of-veterans-affairs/va.gov-team-sensitive/688](https://app.zenhub.com/workspaces/identity-5f5bab705a94c9001ba33734/issues/gh/department-of-veterans-affairs/va.gov-team-sensitive/688)

## Testing done

- ***Previous behavior:***
        sso_logging_info didn't add user session.uuid to logger after direct deposit occurred. 
    
- ***Steps required to verify your changes are working as expected***
        Ran end-to-end  to see if user session.uuid was added to logger after direct deposit occurred. 
                 
- ***Describe the tests completed and the results***
        Rspec test: for def sso_logging_info - test logger for expected result. 
                            
                             

## What areas of the site does it impact?
      Logger
  

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

File: spec/controllers/application_controller_spec.rb [hunk 1]
```diff
@@ -491,4 +496,32 @@ def append_info_to_payload(payload)
...
+      allow(Rails.logger).to receive(:info)
+    end
+
+    context 'when the current user and session object exist' do
+      it 'returns a hash with the user and session info' do
+        expected_result = {
+          user_uuid: user.uuid,
+          sso_cookie_contents: sso_cookie_content, # assuming this method returns a string
+          request_host:
+        }
```
Comment (rileyanderson) [inline spec/controllers/application_controller_spec.rb:518]: Could you put this in a `let`?

File: spec/controllers/application_controller_spec.rb [hunk 2]
```diff
@@ -491,4 +496,32 @@ def append_info_to_payload(payload)
...
+        expect(Rails.logger).to receive(:info) do
+          expect(expected_result).to match({ user_uuid: user.uuid, sso_cookie_contents: 'sso_cookie_contents',
+                                             request_host: 'www.example.com' })
```
Comment (rileyanderson) [inline spec/controllers/application_controller_spec.rb:521]: This actually isn't testing anything. All this is doing is comparing the `expected_result` hash with itself. The idea here would be to test if `expected_result` is actually in the logger. Something like:
`expect(Rails.logger).to receive(:info).with(expected_result)`

**Comment by bosawt** [comment]
I tested by adding `sso_logging_info` in a log in the `UsersController`:
- With Sign in Service authentication, `session_uuid` is now present: 
> { :user_uuid => "ebf2e6ed-53b6-409d-a30a-cc98ae2adc01", :sso_cookie_contents => { "patientIcn" => "1013610747V096981", "signIn" => { "serviceName" => "logingov", "authBroker" => "sis", "clientId" => "vaweb" }, "credential_used" => "logingov", "session_uuid" => "c6d02a11-21aa-4a03-ac60-cfe59daa7039", "expirationTime" => "2023-04-18T21:57:37Z" }, :request_host => "localhost" }

- With SSOe authentication: Looks like I can't test SSOe authentication right now, SSOe isn't currently working on localhost/dev.va.gov stacks

**Comment by bosawt** [comment]
Only feedback I have: Can you update the 'Summary of the PR', this actually updates the `sso_logging_info` function to include the session identifier, which is used by many controllers to intentionally track user behavior

**Review by rileyanderson — APPROVED** [review]
Looks good. Received expected log

**Review by ryan-mcneil — APPROVED** [review]
LGTM! Thanks for the additional test, @rileyanderson!

**Pull request merged**